packages/core/src/components/forms/numericStepper.tsx

```diff
@@ -0,0 +1,413 @@
+            const stepperElems = (buttonPosition === Position.LEFT)
+                ? [buttonGroup, inputGroup]
+                : [inputGroup, buttonGroup];
+
+            return (
+                <div className={stepperClasses}>
+                    {stepperElems}
+                </div>
+            );
+        }
+    }
+
+    public componentWillReceiveProps(nextProps: HTMLInputProps & INumericStepperProps) {
+        super.componentWillReceiveProps(nextProps);
+
+        const value = this.getValueOrEmptyValue(nextProps);
+
+        const didMinChange = nextProps.min !== this.props.min;
+        const didMaxChange = nextProps.max !== this.props.max;
+        const didBoundsChange = didMinChange || didMaxChange;
+
+        // if a new min and max were provided that cause the existing value to fall
+        // outside of the new bounds, then clamp the value to the new valid range.
+        if (didBoundsChange) {
+            const { min, max } = nextProps;
+            const adjustedValue = (value !== NumericStepper.VALUE_EMPTY)
+                ? this.getAdjustedValue(value, /* delta */ 0, min, max)
+                : NumericStepper.VALUE_EMPTY;
+            this.setState({ value: adjustedValue });
+        } else {
+            this.setState({ value });
+        }
+    }
+
+    public componentDidUpdate() {
+        if (this.state.shouldSelectAfterUpdate) {
+            this.inputElement.setSelectionRange(0, this.state.value.length);
+        }
+        this.maybeInvokeOnUpdateCallback(this.state.value);
+    }
+
+    protected validateProps(nextProps: HTMLInputProps & INumericStepperProps) {
+        const { majorStepSize, max, min, minorStepSize, stepSize } = nextProps;
+        if (min && max && min >= max) {
+            throw new Error(Errors.NUMERIC_STEPPER_MIN_MAX);
+        }
+        if (stepSize == null) {
+            throw new Error(Errors.NUMERIC_STEPPER_STEP_SIZE_NULL);
+        }
+        if (stepSize <= 0) {
+            throw new Error(Errors.NUMERIC_STEPPER_STEP_SIZE_NON_POSITIVE);
+        }
+        if (minorStepSize && minorStepSize <= 0) {
+            throw new Error(Errors.NUMERIC_STEPPER_MINOR_STEP_SIZE_NON_POSITIVE);
+        }
+        if (majorStepSize && majorStepSize <= 0) {
+            throw new Error(Errors.NUMERIC_STEPPER_MAJOR_STEP_SIZE_NON_POSITIVE);
+        }
+        if (minorStepSize && minorStepSize > stepSize) {
+            throw new Error(Errors.NUMERIC_STEPPER_MINOR_STEP_SIZE_BOUND);
+        }
+        if (majorStepSize && majorStepSize < stepSize) {
+            throw new Error(Errors.NUMERIC_STEPPER_MAJOR_STEP_SIZE_BOUND);
+        }
+    }
+
+    private renderButton(key: string, iconName: string, onClick: React.MouseEventHandler<HTMLElement>) {
+        const onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+            this.handleButtonKeyDown(e, onClick);
+        };
+        return (
+            <Button
+                disabled={this.props.disabled || this.props.readOnly}
```

Comment:
I don't want the input width to change though - maybe we can extend the width of the `<input>` to the right edge of where the buttons would have been though? even then, we'd have to assume and hardcode the width of the buttons in CSS, which feels a little gross.